packages/core/src/lib/lit-core.ts

```diff
@@ -696,6 +732,44 @@ export class LitCore {
     return res as NodeCommandServerKeysResponse;
   };
 
+  getCurrentEpochNumber = async (): Promise<number> => {
+    try {
+      // (epoch) step 3: first check the cache and use the cached value if it's not more than 30 seconds old. If it's a cache miss or the cached value is too old, fall back to fetching the current epoch number
+      const now = Date.now();
+      const cacheAge = now - this.epochCache.lastUpdateTime!;
+
+      if (
+        this.epochCache.number !== null &&
+        cacheAge <= DELAY_BEFORE_NEXT_EPOCH
```

Comment:
i think this should be separate from DELAY_BEFORE_NEXT_EPOCH and have a different max cache age.  like the DELAY_BEFORE_NEXT_EPOCH is 30s because we want to give the nodes 30s to see the new epoch.  but we can technically cache the current epoch for much longer than that.  the epoch time is 60 mins, and the epoch shouldn't change during that time.  

and since we have the event listener for when the change happens, i'm not sure we actually even need a max cache age here?  what do you think?  

i suppose the purpose of the max cache age is, in case we somehow miss an epoch change event?  but i don't think that should happen often - even if the event polling fails it should pick up the event on the next poll, and that's like a 5s polling interval.  i kinda think we can remove max cache age and handle this corner case in a future PR described below:

we should have error handling in a future PR, to handle the state where the nodes go "hey this epoch is too old!" and the SDK goes "ok let me get and cache the current epoch and retry".  but you can't really write that code until we have the error coming from the Node, because you won't know how to handle it.  so we can save that for a future PR.

Reply:
hmm if we remove the check `cacheAge <= MAX_CACHE_AGE` here and leaving only `this.epochCache.number !== null`, we will have a stale state relying on the epoch that we initially set. 

If we remove the checks altogether, I'm not sure if we still need to have the whole caching mechanism at all? as we are only setting them and not using them. 

I think what we can do is, try to `getCurrentEpochNumber()` from the staking contract, if it fails, we will use the cache value instead. 

☝️should I do this?

Reply:
but the event listener will fire and set this.epochCache.number when the epoch changes?  so it won't be stale, it will always be up to date?

Reply:
Yes, it won't for the first time because `this.epochCache.number` is null, then it will get the epoch number from the contract and set, then the second time the event listener will just return `this.epochCache.number` because it's not null anymore.

Reply:
Let me know if this change works? https://github.com/LIT-Protocol/js-sdk/pull/380/commits/53bacb9f1a2b5eebb2ed358592dcdd00288b27fb @glitch003